content/articles/securing-mongodb-in-linux/index.md

```diff
@@ -0,0 +1,206 @@
+
+```bash
+#note that without closing parenthesis MongoDB will still not run.
+    use admin
+    db.createUser({
+        user: "AndrewKulundeng'",
+        pwd: "xandrew12@12",#enter roles, closing braces and parenthis.
+        role: [{role:"userAdminAnyDatabase", dp:"admin"}]})
+```
+The method executes properly to prompt the user to input password as shown:
+
+```bash
+*enter password used when creating the database, press enters to execute.*
+Enter password: xandrew12@12
+Output
+Successfully added user: {
+     user : "AndrewKulundeng'",
+     roles : [
+        {
+            "role" : "userAdminAnyDatabase",
+            "db" : "admin"
+        }
+      
+    ]
+}
+```
+Then exist MongoDB after the operation. Users will be allowed to enter details after authentication is enabled and the MongoDB daemon restarts. we can as well restart using the command below:
+```bash
+mongod --auth --dbpath="C:\data\mongo" 0r
+#press enters key to execute.
+$sudo systemctl restart mongodb
+```
+### Turn on authentication
+To make authentication to enable, edit `mongod.conf files` as we have done before then restart your mongo service. This will still allow users to access the database without varifying for passwords. They will not be able to make changes to the database since they do not have the correct username and password. Open the configuration file using the command below and move down to the security. Remove the #sign to the security and add authorization colon and set it to enable as shown:
+
+```bash
+usr/etc/mongod.conf
+security:
+```
+Then add authentication parameter and set it to enable. Note that authorization is indented it will look like this:
+
+```bash
+security:
+  authorization: "enabled"
+```
+Close the command line and restart the mongo using the command:
+```bash
+$sudo systemctl restart mongod
+```
+### Authentication Testing
+To test the authentication is working correctly, connect to the database without placing the credentials and check the action of the database you are accessing if it's restricted or not. To check, use the command:
+
+```bash
+$mongo
+```
+No warning will appear since authentication is already enabled. The output will appear similar to this:
+
+```bash
+MongoDB shell version v3.5.0
+```
+The command `show dbs` shows if access is restricted.
+
+```bash
+show dbs
+```
+If there is no message displayed, mean authentication is running as expected. Users will not be able to access the database and modify any other things in the database. Exit the shell by pressing `CTRL + C.` To check if the administrative user authenticates, we run the following commands to connect the user. The `-u` flag for a user to connect to and the `-p` flag requests the user for a password.
+
+```bash
+mongo -u AndrewKulundeng' -p --authentication database admin
+ ```
+Enter user password when prompted. Then issue shows the DBS command to show if the authentication is enabled:

```

Comment:
This sentence "Then issue shows the DBS command to show if the authentication is enabled:" isn't clear.